File: app/src/main/java/ca/usherbrooke/bonpari/api/Player.kt
package ca.usherbrooke.bonpari.api

import com.squareup.moshi.Json
import org.json.JSONObject
import java.io.Serializable

data class Player(
    @Json(name = "prenom") val firstName: String,
    @Json(name = "nom") val lastName: String,
    @Json(name = "age") val age: Int,
    @Json(name = "rang") val rank: Int,
    @Json(name = "pays") val country: String) : Serializable {

    fun getFullName() : String = "$firstName $lastName"

    companion object {
        fun fromJsonObject(jsonObject: JSONObject) : Player {
            return Player(
                jsonObject.getString("prenom"),
                jsonObject.getString("nom"),
                jsonObject.getInt("age"),
                jsonObject.getInt("rang"),
                jsonObject.getString("pays")
            )
        }
    }
}

File: original/website/api/Player.js
"use strict";
class Player {
    constructor(firstName, lastName, age, rank, country) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.age = age;
        this.rank = rank;
        this.country = country;
    }
    static parse(player) {
        return new Player(player.prenom, player.nom, player.age, player.rang, player.pays);
    }
    getFullName() {
        return this.firstName + " " + this.lastName;
    }
}
